cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -217,6 +217,10 @@ type NodeRegistrationOptions struct {
 
 	// IgnorePreflightErrors provides a slice of pre-flight errors to be ignored when the current node is registered.
 	IgnorePreflightErrors []string
+
+	// ImagePullPolicy specifies the policy of pull image. The value of this field is one of Always,IfNotPresent or Never
+	// if this field is unset, i.e. nil, kubeadm will pull the requested images if then not exist on the node.
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
```

Comment:
How about just using `v1.PullPolicy` instead of string here?

Reply:
```suggestion
	ImagePullPolicy v1.PullPolicy
```

- internal types should not have JSON tags.
- use v1.PullPolicy instead of `string`

> How about just using v1.PullPolicy instead of string here?

i think we should avoid API binding between kubeadm types even to Core v1 or Meta v1, if possible, but given we are already embedding v1.Taint, v1.HostPathType and metav1.TypeMeta i think this is fine for v1beta3 and for the time being.

Reply:
Ok, I will change it.